src/client/game.cpp

```diff
@@ -3730,15 +3731,56 @@ bool Game::nodePlacement(const ItemDefinition &selected_def,
 		LocalPlayer *player = client->getEnv().getLocalPlayer();
 
 		// Don't place node when player would be inside new node
-		// NOTE: This is to be eventually implemented by a mod as client-side Lua
-		if (!predicted_f.walkable ||
+		const bool skip_check =  !predicted_f.walkable ||
 				g_settings->getBool("enable_build_where_you_stand") ||
-				(client->checkPrivilege("noclip") && g_settings->getBool("noclip")) ||
-				(predicted_f.walkable &&
-					neighborpos != player->getStandingNodePos() + v3s16(0, 1, 0) &&
-					neighborpos != player->getStandingNodePos() + v3s16(0, 2, 0))) {
+				(client->checkPrivilege("noclip") && g_settings->getBool("noclip"));
+
+		// We are simply colliding the player's current collisionbox in the world to a
+		// "normal node" collision box. This can be improved to take nodeboxes.
+		bool narrow_phase_collision = false;
+		if (!skip_check) {
+			// Collect all node collision boxes.
+			auto cboxes = new std::vector<aabb3f>();
+			predicted_node.getCollisionBoxes(nodedef_manager, cboxes, 0);
```

Comment:
As v-rob said, you don't need to allocate `cboxes` on the heap.

Instead of passing `0` for `neighbors`, you should pass the actual neighbors (`predicted_node.getNeighbors`) so that connected nodeboxes can be resolved correctly.

Reply:
>As v-rob said, you don't need to allocate cboxes on the heap.

Fixed in: 30e50ea2c4759dbdc59a93f98eb21239daf7e56c